### PR TITLE
ref(proguard): Deobfuscate view hierarchies as part of symbolication

### PR DESCRIPTION
**Pull request by loewenheim**
Depends on https://github.com/getsentry/symbolicator/pull/1496.

This moves view hierarchy deobfuscation out of the plugin/preprocessor and into symbolication. That allows us to remove proguard processing from Sentry entirely and only have it in Symbolicator.

Since this means we would potentially want to send JVM events to Symbolicator that don't have exceptions or stacktraces, we have to be more lenient when detecting such events (see the change to `is_jvm_event`). I don't think this presents a problem—we still check if the event contains anything that needs to be symbolicated before we actually send it.

~The bigger problem is that merely modifying the `attachment_cache` the way I'm doing right now doesn't appear to be enough to actually save the modified attachment. The current code where that happens in the preprocessor does just that, but I assume there is some point where the cache is persisted that we don't hit if we do this in symbolication. I need to understand this better.~ This was due to some trivial mistake or other that got fixed along the way.

### CODE REVIEW

File: tests/sentry/ingest/ingest_consumer/test_ingest_consumer_processing.py
Comment (loewenheim) [inline tests/sentry/ingest/ingest_consumer/test_ingest_consumer_processing.py:1]: Changes in this file look much worse than they are. Basically we need to make the test compatible with Symbolicator, so it needs `with set_sentry_option("system.url-prefix", live_server.url):` at the beginning.

File: src/sentry/lang/java/processing.py [hunk 1]
```diff
@@ -139,6 +144,59 @@ def _get_release_package(project: Project, release_name: str | None) -> str | No
...
+    return class_names
+
+
+def _deobfuscate_view_hierarchy(view_hierarchy: Any, class_names: dict[str, str]):
+    windows_to_deobfuscate = [*view_hierarchy.get("windows")]
+
+    while windows_to_deobfuscate:
+        window = windows_to_deobfuscate.pop()
+        if window.get("type") is not None and class_names.get(window["type"]) is not None:
+            window["type"] = class_names.get(window["type"])
```
Comment (Swatinem) [inline src/sentry/lang/java/processing.py:172]:
```suggestion
        if window.get("type") is not None and (mapped_type := class_names.get(window["type"])) is not None:
            window["type"] = mapped_type
```

File: src/sentry/lang/java/processing.py [hunk 2]
```diff
@@ -248,4 +314,8 @@ def process_jvm_stacktraces(symbolicator: Symbolicator, data: Any) -> Any:
         raw_exc["module"] = exc["module"]
         raw_exc["type"] = exc["type"]
 
+    classes = response.get("classes")
+    new_attachments = _deobfuscate_view_hierarchies(attachments, classes)
+    attachment_cache.set(cache_key, attachments=new_attachments, timeout=CACHE_TIMEOUT)
```
Comment (Swatinem) [inline src/sentry/lang/java/processing.py:319]: instead of re-setting *all* the attachments, you could only overwrite the view hierarchy ones.

Comment (loewenheim) [reply src/sentry/lang/java/processing.py:319]: How? I was under the impression I had to set the entire `attachments` list at once here.

**Review by Swatinem — APPROVED** [review]

**Pull request merged**